### PR TITLE
[opt](nereids) make AGG_SCALAR_SUBQUERY_TO_WINDOW_FUNCTION rewrite rule

### PR DESCRIPTION
**Pull request by englefly**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/jobs/executor/Rewriter.java [hunk 1]
```diff
@@ -151,10 +151,8 @@ public class Rewriter extends AbstractBatchJobExecutor {
                     // we need run the following 2 rules to make AGG_SCALAR_SUBQUERY_TO_WINDOW_FUNCTION work
                     bottomUp(new PullUpProjectUnderApply()),
                     topDown(new PushdownFilterThroughProject()),
-                    costBased(
-                            custom(RuleType.AGG_SCALAR_SUBQUERY_TO_WINDOW_FUNCTION,
-                                    AggScalarSubQueryToWindowFunction::new)
-                    ),
+                    custom(RuleType.AGG_SCALAR_SUBQUERY_TO_WINDOW_FUNCTION,
```
Comment (xzj7019) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/jobs/executor/Rewriter.java:154]: it is not always to have positive return for this costbased rule, under our current incomplete code model, we also need a mechanism to control this for on or off.

**Review by jackwener — APPROVED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/jobs/executor/Rewriter.java (hunk 1 above)
Comment (englefly) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/jobs/executor/Rewriter.java:154]: good comments.
we will make framework more agile to support change rules at runtime.

**Review by starocean999 — APPROVED** [review]

**Pull request merged**